rcl/src/rcl/logging_rosout.c

```diff
@@ -290,6 +312,9 @@ void rcl_logging_rosout_output_handler(
       RCUTILS_SAFE_FWRITE_TO_STDERR("\n");
     }
   }
+
+clean:
+  allocator.deallocate(name_dup, allocator.state);
```

Comment:
You could move this deallocation directly after the loop you added above; we never use `name_dup` after that.